pkg/volume/plugins.go

```diff
@@ -688,6 +688,8 @@ func (pm *VolumePluginMgr) FindPluginByName(name string) (VolumePlugin, error) {
 // Check if probedPlugin cache update is required.
 // If it is, initialize all probed plugins and replace the cache with them.
 func (pm *VolumePluginMgr) refreshProbedPlugins() {
+	pm.mutex.Lock()
+	defer pm.mutex.Unlock()
```

Comment:
ok... at the point where we're making non-test changes, I think we should revert the original PR to get back to the previous state and come in again fresh with a full review of correctness

Reply:
reopened revert PR https://github.com/kubernetes/kubernetes/pull/128450